sympy/ntheory/partitions_.py

```diff
@@ -1,40 +1,98 @@
 from __future__ import print_function, division
-
 from mpmath.libmp import (fzero,
     from_man_exp, from_int, from_rational,
     fone, fhalf, bitcount, to_int, to_str, mpf_mul, mpf_div, mpf_sub,
-    mpf_add, mpf_sqrt, mpf_pi, mpf_cosh_sinh, pi_fixed, mpf_cos)
+    mpf_add, mpf_sqrt, mpf_pi, mpf_cosh_sinh, pi_fixed, mpf_cos, mpf_sin)
 from sympy.core.numbers import igcd
 import math
 from sympy.core.compatibility import range
+from .residue_ntheory import (_sqrt_mod_prime_power,
+    legendre_symbol, jacobi_symbol, is_quad_residue)
+
+maxn = 10**5 + 5
+factor = [0]*maxn  #For storing the smallest prime factor
+lim = int(maxn**0.5) + 5
+for i in range(2, lim):
+    if factor[i] == 0:
+        for j in range(i*i, maxn, i):
+            if factor[j] == 0:
+                factor[j] = i
+
+_totient = [1]*maxn
+for i in range(2, maxn):
+    if factor[i] == 0:
+        factor[i] = i
+        _totient[i] = i-1
+    x = factor[i]
+    y = i//x
+    if y%x == 0:
+        _totient[i] = _totient[y]*x
+    else:
+        _totient[i] = _totient[y]*(x - 1)
```

Comment:
This initialization part should probably be made a function, e.g. ``_init_tables``, that would be called by ``npartitions``. ``_totient`` and ``factor`` (perhaps ``_factor`` would be preferable) are initially set to ``None``. When ``npartitions`` is called (perhaps with n > 5), ``_totient`` is tested, and ``_init_tables`` is run if it is undefined. (It may be necessary to define ``_totient`` global in ``npartitions``.)